Sources/Sentry/SentryClient.m

```diff
@@ -63,6 +63,16 @@ - (_Nullable instancetype)initWithOptions:(SentryOptions *)options
         self.threadInspector =
             [[SentryThreadInspector alloc] initWithStacktraceBuilder:stacktraceBuilder
                                             andMachineContextWrapper:machineContextWrapper];
+
+        NSError *error = nil;
+        self.fileManager = [[SentryFileManager alloc] initWithDsn:self.options.parsedDsn
+                                                 didFailWithError:&error];
+        if (nil != error) {
+            [SentryLog logWithMessage:error.localizedDescription andLevel:kSentryLogLevelError];
+        }
+
+        self.transport = [SentryTransportFactory initTransport:self.options
+                                             sentryFileManager:self.fileManager];
```

Comment:
H: not sure if this is a good thing, while we always need a transport, one may have their own transport implementation and they would like to use that one, this change seems to kill this possibility.
Actually the options should have a `transport` field which is set by the client, but I failed to find it.
https://docs.sentry.io/error-reporting/configuration/?platform=csharp#transport-options

Reply:
That's a good point and we are aware that this needs to be fixed. Currently, you can't set the transport to the options and the transport here is not public. So this change doesn't have an impact on this problem. I would like to fix this, but definitely not in this PR.

Reply:
It's even tricker now a days since we have our Rate Limit implementation used by the transport and that's def something we don't want to be swappable.
We could have a decorator transport, where the decorator itself is not public/swappable and what we let the user change is the last in chain, or else. 

Swapping transports is also more or less relevant depending on the tech. For iOS/Android it seems people are usually fine with what's in the box. For native and JS for example people seem to have more need to add a custom one

Reply:
that's a good idea, I'd like to do the same on Android/Java, not needed in this PR but maybe a new task to not forget about it.